src/components/SeatMap/service.test.js

```diff
@@ -0,0 +1,1156 @@
+
+      expect(response).toEqual({
+        content: mockContent,
+        params: mockParams,
+        exits: mockExits,
+        bulks: mockBulks,
+        availabilityData: mockPlaneFeatures.availabilityData,
+      });
+
+      expect(mockGetPlaneFeatures).toHaveBeenCalledTimes(1);
+      expect(mockGetPlaneFeatures).toHaveBeenCalledWith(flight, config.lang, config.units);
+      expect(mockPrepareData).toHaveBeenCalledTimes(1);
+      expect(mockPrepareData).toHaveBeenCalledWith(mockPlaneFeatures, config);
+      expect(mockSetAvailabilityHandler).toHaveBeenCalledTimes(1);
+      expect(mockSetAvailabilityHandler).toHaveBeenCalledWith(mockContent, availability);
+      expect(mockSetPassengersHandler).not.toHaveBeenCalled();
+    });
+  });
+
+  describe('selectSeatHandler', () => {
+    it('should set passenger seat, price & number for next passenger', () => {
+      const service = createSeatsMapService();
+
+      const passengerId = '12345';
+      const passenger = {
+        id: passengerId,
+      };
+      const getNextPassengerMock = jest.spyOn(service, 'getNextPassenger').mockImplementation(() => passenger);
+
+      const setPassengersHandlerReturnValue = {
+        key: 'value',
+      };
+      const setPassengersHandlerMock = jest
+        .spyOn(service, 'setPassengersHandler')
+        .mockImplementation(() => setPassengersHandlerReturnValue);
+
+      const seat = {
+        passenger: {
+          id: passengerId,
+        },
+        price: '$ 9,99',
+        number: '33A',
+      };
+      const passengersList = [
+        {
+          id: passengerId,
+        },
+      ];
+
+      const { data: actualData, passengers: actualPassengers } = service.selectSeatHandler({}, seat, passengersList);
+
+      expect(actualData).toEqual(setPassengersHandlerReturnValue);
+
+      const expectedPassengers = [
+        {
+          id: passengerId,
+          seat: {
+            price: '$ 9,99',
+            seatLabel: '33A',
+          },
+        },
+      ];
+      expect(actualPassengers).toEqual(expectedPassengers);
+      expect(getNextPassengerMock).toHaveBeenCalledTimes(1);
+      expect(setPassengersHandlerMock).toHaveBeenCalledTimes(1);
+      expect(setPassengersHandlerMock).toHaveBeenCalledWith({}, expectedPassengers);
+    });
+  });
+
+  describe('unselectSeatHandler', () => {
+    it('should clear the seat from any passengers assigned to seats before calling setPassengersHandler', () => {
+      const service = createSeatsMapService();
+
+      const setPassengersHandlerReturnValue = {
+        key: 'value',
+      };
+      const setPassengersHandlerMock = jest
+        .spyOn(service, 'setPassengersHandler')
+        .mockImplementation(() => setPassengersHandlerReturnValue);
+
+      const passengerId = '12345';
+      const seat = {
+        passenger: {
+          id: passengerId,
+        },
+      };
+      const passengersList = [
```

Comment:
Could we have more than one passenger here to ensure the seat is just cleared up to the correct passenger please?

Reply:
Thank you!